msal/managed_identity.py

```diff
@@ -274,6 +274,7 @@ def acquire_token_for_client(self, *, resource):  # We may support scope in the
                     "expires_in": int(expires_in),  # OAuth2 specs defines it as int
                 }
                 if "refresh_on" in entry and int(entry["refresh_on"]) < now:  # aging
+                    access_token_from_cache["refresh_on"] = int(entry["refresh_on"])
                     break  # With a fallback in hand, we break here to go refresh
```

Comment:
Should break up the conditional like you did in `application.py`?
```suggestion
                if "refresh_on" in entry:
                    access_token_from_cache["refresh_on"] = int(entry["refresh_on"])
                    if int(entry["refresh_on"]) < now:  # aging
                        break  # With a fallback in hand, we break here to go refresh
```